keps/sig-instrumentation/3077-contextual-logging/README.md

```diff
@@ -0,0 +1,997 @@
+  dev@kubernetes.io that a logger must be set with k8s.io/klogr.
+- Issues filed against kubernetes-sigs projects with a request to set a logger
+  with k8s.io/klogr. These issues will have a link to the enhancement issue
+  (for cross-referencing) and a link to the developer documentation.
+
+#### GA
+
+- All code in kubernetes/kubernetes converted to contextual logging,
+  no dependency on klog anymore
+- All code under kubernetes-sigs sets a logger with k8s.io/klogr
+  and therefore is ready for the next k/k release.
+- Gather feedback from developers and surveys
+- Allowing time for feedback
+
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+
+### Upgrade / Downgrade Strategy
+
+Not applicable. The log output will be determined by what is implemented in the
+code that currently runs.
+
+### Version Skew Strategy
+
+Not applicable. The log output format is the same as before, therefore other
+components are not affected.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+This is not a feature in the traditional sense. Converted log calls will always
+use contextual logging, the only supported rollback is to downgrade to a
+previous release.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Other
+  - Describe the mechanism: updating to a new release where code was converted.
+  - Will enabling / disabling the feature require downtime of the control
+    plane? Yes, because components must be restarted.
+  - Will enabling / disabling the feature require downtime or reprovisioning of
+    a node? kubelet must be restarted, but the node itself can continue to run.
+
+###### Does enabling the feature change any default behavior?
+
+No. Unless log messages get intentionally enhanced as part of touching the
+code, the log output will be exactly the same as before.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+No.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Previous state is irrelevant, so a previous rollback has no effect.
+
+###### Are there any tests for feature enablement/disablement?
+
+Not applicable.
+
+### Rollout, Upgrade and Rollback Planning
+
+Nothing special needed. The same logging flags as before will be supported.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The worst case would be that a null logger instance somehow gets passed into a
+function and then causes log calls to crash. The design of the APIs makes that
+very unlikely and code reviews should be able to catch the code that causes
+this.
+
+###### What specific metrics should inform a rollback?
+
+Components start to crash with a nil pointer panic in the `logr` package.
```

Comment:
Rollback would be a full major version for this. We will need to be good about cherry-picks if we find anything like this.